### PR TITLE
Jarvis devices

### PR DESCRIPTION
**Pull request by alexcherman**

### CODE REVIEW

File: universal-login-wallet/src/core/presenters/WalletPresenter.ts [hunk 1]
```diff
@@ -5,6 +5,16 @@ export class WalletPresenter {
...
+  getPrivateKey() {
+    switch (this.walletService.state.kind) {
+      case 'Connecting':
+      case 'Deployed':
+        return this.walletService.state.wallet.privateKey;
+      default:
+        throw new Error('Invalid wallet state: expected Deployed or Connecting wallet');
+    }
+  }
+
```
Comment (alexcherman) [inline universal-login-wallet/src/core/presenters/WalletPresenter.ts]: Not sure about it. Don't know what is the difference between 'Connecting' and 'Deployed'. Made similarly to getName()

Comment (marekkirejczyk) [reply universal-login-wallet/src/core/presenters/WalletPresenter.ts]: That is a bit unintuitive indeed. It is fine for now.

In the future, we should sort out WalletService states better.
Perhaps `{Future, Deployed, Connected, Deploying, Connecting}`

**Comment by marekkirejczyk** [comment]
Needs rebase @alexcherman

**Comment by marekkirejczyk** [comment]
@alexcherman @xitronix

**Comment by alexcherman** [comment]
@marekkirejczyk rebased

**Pull request merged**